scenes/globals/game_state/game_state.gd

```diff
@@ -37,8 +41,25 @@ func _ready() -> void:
 	_restore()
 
 
-func start_quest() -> void:
-	clear_inventory()
+func start_quest(quest: Quest) -> void:
+	_do_clear_inventory()
+	_update_inventory_state()
+	_state.set_value(QUEST_SECTION, QUEST_PATH_KEY, quest.resource_path)
```

Comment:
I think this is a useful thing to have in the (persisted) game state in order to implement "abandon quest".